hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -819,6 +825,14 @@ public IOException call() {
             } else {
               effectiveVolumes.add(volume.toString());
               LOG.info("Successfully added volume: {}", volume);
+              if (errorDisk != null && !errorDisk.isEmpty()) {
```

Comment:
It may be not necessary to check if `errorDisk` is empty.

Reply:
I don't think so. If `refreshVolumes` is invoked by `reconfigurePropertyImpl`, `errorDisk` may be null.

``` java
  public String reconfigurePropertyImpl(String property, String newVal)
      throws ReconfigurationException {
    switch (property) {
      case DFS_DATANODE_DATA_DIR_KEY: {
        IOException rootException = null;
        try {
          LOG.info("Reconfiguring {} to {}", property, newVal);
          this.refreshVolumes(newVal);
...
  }
```